### PR TITLE
Update dependencies.

### PR DESCRIPTION
**Pull request by wsanchez**
Update dependencies.

Note all updates here are to `tox.ini`, not `setup.py`, so it primarily effects testing.

This also updates `flake8` and `mypy` runs to use Python 3.7, and adds Python 3.7 to the test matrix  and changes the Travis Linux distribution to Xenial, which appears to be the Latest Hotness.

### CODE REVIEW

**Comment by wsanchez** [comment]
Gonna leave out the `mypy` update, as I still need to figure out a new error; will do in a separate PR.

**Comment by wsanchez** [comment]
`mypy` issue filed with `attrs`: https://github.com/python-attrs/attrs/issues/474

…and it's a bug fixed in `mypy`, due in next release: https://github.com/python/mypy/pull/6006

File: .gitignore [hunk 1]
```diff
@@ -3,6 +3,7 @@
 .DS_Store
 /.coverage
 /.coverage.*
+/.eggs
```
Comment (wsanchez) [inline .gitignore]: This is opportunistic

File: .travis.yml [hunk 2]
```diff
@@ -1,5 +1,4 @@
-dist: trusty
-sudo: false
+dist: xenial
```
Comment (wsanchez) [inline .travis.yml]: Required to add Python 3.7

File: .travis/mypy [hunk 3]
```diff
@@ -14,23 +14,28 @@ tmp="$(mktemp -t mypy.XXXX)";
 #
 # Notes:
 #
-# * mypy notices trial's use of a skip marker attribute on functions and
-#   complains that it's not part of a function object's interface, so we
-#   ignore: has no attribute "skip". Same with "todo".
-#
```
Comment (wsanchez) [inline .travis/mypy]: This filter was removed previously; comment no longer applies.

File: src/klein/test/_strategies.py [hunk 4]
```diff
@@ -12,8 +12,9 @@
 
 from hyperlink import DecodedURL, EncodedURL
 
+from hypothesis import assume
 from hypothesis.strategies import (
-    assume, characters, composite, integers, lists, sampled_from, text
+    characters, composite, integers, lists, sampled_from, text
 )
```
Comment (wsanchez) [inline src/klein/test/_strategies.py]: Import assume from correct location.  This one fails now.

File: src/klein/test/test_trial.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
+# Copyright (c) 2017-2018. See LICENSE for details.
+
+"""
+Tests for L{klein.test._trial}.
```
Comment (wsanchez) [inline src/klein/test/test_trial.py]: Added due to lack of coverage.

File: tox.ini [hunk 6]
```diff
@@ -4,8 +4,7 @@ envlist =
     flake8
     mypy
 
-    # Twisted 15.5 is the first version to support Python 3 (3.3+)
-    coverage-py{27,34,35,36,py2,py3}-tw{166,171,current,trunk}
+    coverage-py{27,34,35,36,37,py2,py3}-tw{166,171,current,trunk}
```
Comment (wsanchez) [inline tox.ini]: Comment moved to setup.py, where it is relevant.
py37 added to default env list.

**Comment by glyph** [comment]
> Xenial, which appears to be the Latest Hotness

Xenial was the Latest Hotness in … April of 2016.  (Still, latester and hottester than Trusty, which was the Latest Hotness in *2014*, so happy to upgrade.)

**Pull request merged**